ManusGloves/include/ManusGloveHelper.h

```diff
@@ -0,0 +1,377 @@
+     * In order to have any 3d positional/rotational information from the gloves or body,
+     * one needs to setup a skeleton on which this data can be applied.
+     * In the case of this sample we create a Hand skeleton in order to get skeleton information
+     * in the OnSkeletonStreamCallback function. This sample does not contain any 3D rendering, so
+     * we will not be applying the returned data on anything.
+     */
+    void LoadSkeleton();
+
+    /**
+     * Skeletons are pretty extensive in their data setup
+     * so we have several support functions so we can correctly receive and parse the data, 
+     * this function helps setup the data.
+     * @param p_Id the id of the created node setup
+     * @param p_ParentId the id of the node parent
+     * @param p_PosX X position of the node, this is defined with respect to the global coordinate system or the local one depending on 
+     * the parameter p_UseWorldCoordinates set when initializing the sdk,
+     * @param p_PosY Y position of the node this is defined with respect to the global coordinate system or the local one depending on 
+     * the parameter p_UseWorldCoordinates set when initializing the sdk,
+     * @param p_PosZ Z position of the node this is defined with respect to the global coordinate system or the local one depending on 
+     * the parameter p_UseWorldCoordinates set when initializing the sdk,
+     * @param p_Name the name of the node setup
+     * @return the generated node setup
+     */
+    NodeSetup CreateNodeSetup(uint32_t p_Id, uint32_t p_ParentId, float p_PosX, float p_PosY, float p_PosZ, std::string p_Name);
+
+    /**
+     * 
+     */
+    static ManusVec3 CreateManusVec3(float p_X, float p_Y, float p_Z);
+
+protected:
+    static ManusGloveHelper *s_Instance;
+    bool m_Running = true;
+    std::string ManusGlove_LogPrefix = "ManusGloveHelper::";
+
+    std::function<ClientReturnCode()> m_CurrentInteraction = nullptr;
+
+    // Networking
+    bool m_ShouldConnectLocally = true;
+
+    uint32_t m_HostToConnectTo = 0;
+    uint32_t m_NumberOfHostsFound = 0;
+    uint32_t m_SecondsToFindHosts = 2;
+
+    int32_t m_SecondsToAttemptReconnecting = 60;
+    int32_t m_MaxReconnectionAttempts = 10;
+    uint32_t m_SleepBetweenReconnectingAttemptsInMs = 100;
+
+    std::unique_ptr<ManusHost[]> m_AvailableHosts = nullptr;
+    std::unique_ptr<ManusHost> m_Host;
+
+    //Data
+    uint32_t m_SessionId = 0;
+
+    int t_DataOffset = 0;
+    int j_DataOffset = 0;
+    std::shared_ptr<int> t_counter = std::make_shared<int>(0);
+
+    std::vector<uint32_t> m_LoadedSkeletons;
+
+    std::chrono::time_point<std::chrono::high_resolution_clock> m_TimeSinceLastDisconnect;
+
+    std::mutex m_SkeletonMutex;
+
+    ClientSkeletonCollection *m_NextSkeleton = nullptr;
+    ClientSkeletonCollection *m_Skeleton = nullptr;
+
+    std::mutex m_LandscapeMutex;
+    Landscape *m_NewLandscape = nullptr;
+    Landscape *m_Landscape = nullptr;
```

Comment:
Why not use a smart pointer?

Reply:
> Why not use a smart pointer?

Actually, the skeleton ones are not required so I removed them.
But in general, while there is a possibility of using smart pointers, I wanted to stick to the official guide to have a working drive as fast as possible.

Reply:
It makes sense. Compatibly with the your deadlines, it may also make sense to gradually switch to a more modern pattern anyhow

Reply:
you are absolutely right. I will try to implement it by Monday.